receiver/couchdbreceiver/metrics.go

```diff
@@ -0,0 +1,215 @@
+	methods := []string{metadata.AttributeHTTPMethod.COPY, metadata.AttributeHTTPMethod.DELETE, metadata.AttributeHTTPMethod.GET, metadata.AttributeHTTPMethod.HEAD, metadata.AttributeHTTPMethod.OPTIONS, metadata.AttributeHTTPMethod.POST, metadata.AttributeHTTPMethod.PUT}
+	for _, method := range methods {
+		httpdRequestMethodKey := []string{"httpd_request_methods", method, "value"}
+		httpdRequestMethodValue, err := getIntFromBody(httpdRequestMethodKey, stats)
+		if err != nil {
+			c.logger.Info(
+				err.Error(),
+				zap.String("key", strings.Join(httpdRequestMethodKey, " ")),
+			)
+			return
+		}
+
+		c.mb.RecordCouchdbHttpdRequestsDataPoint(now, httpdRequestMethodValue, method)
+	}
+}
+
+func (c *couchdbScraper) recordCouchdbHttpdResponsesDataPoint(now pdata.Timestamp, stats map[string]interface{}) {
+	codes := []string{"200", "201", "202", "204", "206", "301", "302", "304", "400", "401", "403", "404", "405", "406", "409", "412", "413", "414", "415", "416", "417", "500", "501", "503"}
+	for _, code := range codes {
+		httpdResponsetCodeKey := []string{"httpd_status_codes", code, "value"}
+		httpdResponsetCodeValue, err := getIntFromBody(httpdResponsetCodeKey, stats)
+		if err != nil {
+			c.logger.Info(
+				err.Error(),
+				zap.String("key", strings.Join(httpdResponsetCodeKey, " ")),
+			)
+			return
+		}
+
+		c.mb.RecordCouchdbHttpdResponsesDataPoint(now, httpdResponsetCodeValue, code)
+	}
+}
+
+func (c *couchdbScraper) recordCouchdbHttpdViewsDataPoint(now pdata.Timestamp, stats map[string]interface{}) {
+	views := []string{metadata.AttributeView.TemporaryViewReads, metadata.AttributeView.ViewReads}
+	for _, view := range views {
+		httpdResponsetCodeKey := []string{"httpd", view, "value"}
+		httpdResponsetCodeValue, err := getIntFromBody(httpdResponsetCodeKey, stats)
+		if err != nil {
+			c.logger.Info(
+				err.Error(),
+				zap.String("key", strings.Join(httpdResponsetCodeKey, " ")),
+			)
+			return
+		}
+
+		c.mb.RecordCouchdbHttpdViewsDataPoint(now, httpdResponsetCodeValue, view)
+	}
+}
+
+func (c *couchdbScraper) recordCouchdbDatabaseOpenDataPoint(now pdata.Timestamp, stats map[string]interface{}) {
+	openDatabaseKey := []string{"open_databases", "value"}
+	openDatabaseMetricValue, err := getIntFromBody(openDatabaseKey, stats)
+	if err != nil {
+		c.logger.Info(
+			err.Error(),
+			zap.String("key", strings.Join(openDatabaseKey, " ")),
+		)
+		return
+	}
+
+	c.mb.RecordCouchdbDatabaseOpenDataPoint(now, openDatabaseMetricValue)
+}
+
+func (c *couchdbScraper) recordCouchdbFileDescriptorOpenDataPoint(now pdata.Timestamp, stats map[string]interface{}) {
+	fileDescriptorKey := []string{"open_os_files", "value"}
+	fileDescriptorMetricValue, err := getIntFromBody(fileDescriptorKey, stats)
+	if err != nil {
+		c.logger.Info(
+			err.Error(),
+			zap.String("key", strings.Join(fileDescriptorKey, " ")),
+		)
+		return
+	}
+
+	c.mb.RecordCouchdbFileDescriptorOpenDataPoint(now, fileDescriptorMetricValue)
+}
```

Comment:
I think the error handling is making the code difficult to manage. The `scrapererrors` package will simplify things for you quite a bit:
```suggestion
func (c *couchdbScraper) recordCouchdbFileDescriptorOpenDataPoint(now pdata.Timestamp, stats map[string]interface{}, errors scrapererror.ScrapeErrors) {
	fileDescriptorKey := []string{"open_os_files", "value"}
	fileDescriptorMetricValue, err := getIntFromBody(fileDescriptorKey, stats)
	if err != nil {
		errors.AddPartial(1, err)
		return
	}
	c.mb.RecordCouchdbFileDescriptorOpenDataPoint(now, fileDescriptorMetricValue)
}
```

Reply:
cool, added partial errors